### PR TITLE
Convert median.F90 to C++ nth_element algorithm

### PR DESCRIPTION
**Pull request by jrood-nrel**
Also removes unused files. This is the first step to removing Fortran. I've checked the performance for Tioga in the exawind-driver sphere problem is similar to the original time using `median.F90`. I've also checked (when using `-np 1`) that results are exactly the same in the sphere problem using `exodiff` and `fcompare`.

### CODE REVIEW

**Comment by ashesh2512** [comment]
@jrood-nrel Would it be feasible to test with something like a turbine, i.e. something that isn’t a symmetric geometry ?

File: src/buildADTrecursion.c [hunk 1]
```diff
@@ -48,7 +46,8 @@ void buildADTrecursion(double *coord,double *adtReals,double *adtWork,int *adtIn
     // reorder elements with nleft elements to
     // the left of median of adtWork
     //
-    median_(elementsAvailable,adtWork,&nav,&coordmid);
+    std::sort(elementsAvailable, elementsAvailable + nav,
```
Comment (ashesh2512) [inline src/buildADTrecursion.c]: Appears as though median_ is changing both elementsAvailable and adtWork ? But the new algorithm only changes elementsAvailable ?

**Review by marchdf — APPROVED** [review]

File: src/buildADTrecursion.c (hunk 1 above)
Comment (marchdf) [reply src/buildADTrecursion.c]: are you looking at the right commit? I don't see this diff in the final version

Comment (jrood-nrel) [reply src/buildADTrecursion.c]: Yeah it's weird it's showing the sort call here, but yeah let me look at this again. Yeah this has indexes as well, so I can sort both.

Comment (jrood-nrel) [reply src/buildADTrecursion.c]: Ok so the comparison needs to happen on `adtWork`, but `elementsAvailable` is what needs to be reordered. So `adtWork` is only used for the sort and not after. So I used `adtWork` for the comparison.

**Comment by jrood-nrel** [comment]
So running with anything besides `mpirun -np 1` such as the turbine will show diffs even if you run the original code twice since it's nondeterministic.

File: src/buildADTrecursion.C [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    //
+    if (side > 0) {
+      adtIntegers[4 * parent + side] = elementsAvailable[nleft - 1];
+    }
+    parentToChild = *adtCount;
+    //
+    // build the left side of the tree
+    //
+    if (nleft > 1) {
+      buildADTrecursion(coord, adtReals, adtWork, adtIntegers,
```
Comment (ashesh2512) [inline src/buildADTrecursion.C:93]: @jrood-nrel adtWork is used here and below too.

Comment (jrood-nrel) [reply src/buildADTrecursion.C:93]: It's overwritten entirely each call to `buildADTrecursion` so sorting it looks unnecessary.

**Comment by jrood-nrel** [comment]
I'm going to close this and test it more.

**Pull request closed without merging**